e2e/testcases/override_resource_limits_test.go

```diff
@@ -233,16 +243,18 @@ func TestOverrideReconcilerResourcesV1Alpha1(t *testing.T) {
 
 	// Verify the resource limits of ns-reconciler-backend are not affected by the resource limit change of root-reconciler
 	err = nt.Validate(core.NsReconcilerName(backendNamespace, configsync.RepoSyncName), v1.NSConfigManagementSystem, &appsv1.Deployment{},
-		nomostest.HasCorrectResourceRequestsLimits(reconcilermanager.Reconciler, resource.MustParse("500m"), resource.MustParse("555m"), resource.MustParse("500Mi"), resource.MustParse("555Mi")),
-		nomostest.HasCorrectResourceRequestsLimits(reconcilermanager.GitSync, resource.MustParse("600m"), resource.MustParse("666m"), resource.MustParse("600Mi"), resource.MustParse("666Mi")))
+		hasGeneration(nsReconcilerBackendDeployment.Generation+1),
```

Comment:
I would suggest declaring variables for these generations to show incremental changes. For example, we can have a `nsBackendDeploymentGeneration` and increase the value when needed. In this case, we could clearly see the generation remains the same.

Reply:
Great suggestion! Fixed it. Thanks!